### PR TITLE
Added sha3_224/256/384/512

### LINKED ISSUE #8: Support the multiple SHA3 variants
https://github.com/jbenet/go-multihash/blob/master/multihash.go#L63

Ethereum, for example, uses 32 byte, and is probably the most important user of the function in the immediate future. Maybe best to do `SHA3_256` and `SHA3_512`?

### PR DESCRIPTION
**Pull request by tabrath**
Closes #8.

--
Added closes tag to main comment ~Kubuxu

### CODE REVIEW

File: multihash.go [hunk 1]
```diff
@@ -36,7 +36,11 @@ const (
 	SHA1       = 0x11
 	SHA2_256   = 0x12
 	SHA2_512   = 0x13
-	SHA3       = 0x14
+	SHA3_224   = 0x17
+	SHA3_256   = 0x16
+	SHA3_384   = 0x15
+	SHA3_512   = 0x14
+	SHA3       = SHA3_512
```
Comment (whyrusleeping) [inline multihash.go]: :+1: Great to be explicit on this :)

**Comment by whyrusleeping** [comment]
@tabrath mind rebasing? The changes here contain the murmur and keccak changes too

File: multihash.go (hunk 1 above)
Comment (tabrath) [inline multihash.go]: Was hoping to keep compatibility with libraries using SHA3..

**Comment by tabrath** [comment]
Closes #8.

**Comment by tabrath** [comment]
@Kubuxu Rebased this too :)

**Pull request merged**